docs/templating/README.md

```diff
@@ -0,0 +1,108 @@
+# Template Hierarchy
+
+Arguably, one of the most powerful systems in WordPress' Theme API is its
+[Template Hierarchy](https://developer.wordpress.org/themes/basics/template-hierarchy/).
+Template Hierarchy is WordPress' way of allowing customizations of various page types on highly dynamic websites while
+also allowing fine granularity to make templating exceptions.
+
+### Template Hierarchy Visual Overview
+
+In the overview below, if you start with a page on the left and trace it to the right, you'll see what template files
+WordPress will attempt to load. If there are no templates found, it will fallback to `index.php`.
+
+![WordPress Template Hierarchy Diagram](https://developer.wordpress.org/files/2014/10/Screenshot-2019-01-23-00.20.04.png)
+(Credit: https://developer.wordpress.org/themes/basics/template-hierarchy/#visual-overview)
+
+## Template Hierarchy in Next.js
+
+While Next.js has a robust [routing layer](https://nextjs.org/docs/routing/introduction), it is not ideal for
+WordPress-based sites as the routes are built around directory and file structures rather than content types.
```

Comment:
I'm not sure if we should explicitly say that Next.js's routing is not ideal for WordPress sites. How about:

> it is not always ideal for WordPress sites

Reply:
Great callout, I'll tweak this